pkg/node/local_node_store.go

```diff
@@ -60,44 +43,53 @@ type LocalNodeStoreParams struct {
 	Init      LocalNodeInitializer `optional:"true"`
 }
 
-// localNodeStore implements the LocalNodeStore using a simple in-memory
-// backing. Reflecting the new state to persistent stores, e.g. kvstore or k8s
-// is left to observers.
-type localNodeStore struct {
+// LocalNodeStore is the canonical owner for the local node object and provides
+// a reactive API for observing and updating the state.
+type LocalNodeStore struct {
+	// Changes to the local node are observable.
 	stream.Observable[LocalNode]
 
-	mu   lock.Mutex
-	cond *sync.Cond
-
-	valid    bool
+	mu       lock.Mutex
 	value    LocalNode
 	emit     func(LocalNode)
 	complete func(error)
 }
 
-var _ LocalNodeStore = &localNodeStore{}
+func newTestLocalNodeStore() *LocalNodeStore {
```

Comment:
nit: Does this need to be in the prod code files or can we put it into the test code files?